File: typescript/week-06-objects/solutions.ts
// === Woche 6: Objekte ===

interface Person {
  name: string;
  age: number;
}

interface Team {
  name: string;
  members: string[];
  count: number;
}

/**
 * Erstellt ein Person-Objekt mit name und age.
 */
export function createPerson(name: string, age: number): Person {
  // TODO: Gib ein Objekt { name, age } zurück
}

/**
 * Gibt "Hello, {name}!" basierend auf dem person-Objekt zurück.
 */
export function greetPerson(person: { name: string }): string {
  // TODO: Gib "Hello, " + person.name + "!" zurück
}

/**
 * Erhöht das Alter der Person um 1 und gibt ein neues Objekt zurück.
 */
export function birthday(person: { age: number }): { age: number } {
  // TODO: Gib ein neues Objekt mit age + 1 zurück
}

/**
 * Gibt true zurück, wenn die Person 65 oder älter ist.
 */
export function isSenior(person: { age: number }): boolean {
  // TODO: Gib true zurück wenn age >= 65
}

/**
 * Erstellt ein Team-Objekt mit name, members und count.
 */
export function createTeam(name: string, members: string[]): Team {
  // TODO: Gib { name, members, count: members.length } zurück
}

File: rust/week_06_structs/src/solutions.rs
// TODO: Füge die Felder name: String und age: u8 hinzu
pub struct Person {
    pub name: String,
    pub age: u8,
}

impl Person {
    // TODO: Erstelle eine neue Person mit name und age
    pub fn new(_name: &str, _age: u8) -> Person {
        todo!()
    }
}

// TODO: Gib "Hello, {name}!" zurück
pub fn greet_person(_person: &Person) -> String {
    todo!()
}

// TODO: Erhöhe age um 1
pub fn birthday(_person: &mut Person) {
    todo!()
}

// TODO: Gib true zurück, wenn age >= 65
pub fn is_senior(_person: &Person) -> bool {
    todo!()
}

// TODO: Füge die Felder name: String, members: Vec<String> und count: usize hinzu
pub struct Team {
    pub name: String,
    pub members: Vec<String>,
    pub count: usize,
}

impl Team {
    // TODO: Erstelle ein neues Team (count = members.len())
    pub fn new(_name: &str, _members: Vec<String>) -> Team {
        todo!()
    }
}
